twisted/internet/test/test_await.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+A wrapper for L{twisted.internet.test._awaittests}, as that test module
+includes keywords not valid in Pythons before 3.5.
+"""
+
+try:
+    from twisted.python.compat import execfile
+    from twisted.python.filepath import FilePath
+
+    _path = FilePath(__file__).parent().child("_awaittests.py.3only")
+
+    _g = {}
+    execfile(_path.path, _g)
+
+    AwaitTests = _g["AwaitTests"]
+    __all__ = ["AwaitTests"]
+except:
```

Comment:
drive-by comment!

This is too broad, in that weird misconfigurations will raise weird errors and these tests silently won't run.

1. Move the `compat` and `FilePath` imports, and the `_path` construction, out of the `try` block. Even the `_g = {}`.  Presumably the only thing that should fail is the `execfile`, so that's the only thing that should be in the `try`.
2. Catch only the exception that's the problem (`SyntaxError`, I'm guessing?) not bare-word except.
3. Even the silly obvious can't fail stuff after the `execfile` should be in the `else` suite of a `try`/`except`/`else`; you don't want to swallow any more errors than you have to.
4. Consider adding an `AwaitTests` class that just has a single, empty test method and a "skip" attribute, so there's at least visibility that the test was there, and it was intentionally not run.
5. Maybe don't even have a `try`.  This is *always* supposed to work on 3.5, right?  So just compare `sys.version_info` and do it unconditionally.

This comment brought to you by literally dozens of saturday nights ruined by `NameError` or `AttributeErrors`s in some obscure `try` block that "would only fail in this one case".